unstable/wlr-exec-secure-unstable-v1.xml

```diff
@@ -0,0 +1,139 @@
+      Clients which want the compositor to execute another client should create
+      a zwlr_exec_handle_v1 and use it to configure and run the client.
+    </description>
+
+    <request name="exec_process">
+      <description summary="creates a wlr_exec_handle_v1 resource">
+        Creates an exec_handle for the requested command. The compositor can
+        choose to disallow this request for certain commands, in which case a
+        protocol error is issued. The command will be executed without a shell.
+      </description>
+      <arg name="handle" type="new_id" interface="zwlr_exec_handle_v1"/>
+      <arg name="command" type="array" summary="the command to execute"/>
+    </request>
+
+    <request name="destroy" type="destructor">
+      <description summary="destroy the manager">
+        All objects created by the manager will still remain valid, until their
+        appropriate destroy request has been called.
+      </description>
+    </request>
+  </interface>
+
+  <interface name="zwlr_exec_handle_v1" version="1">
+    <description summary="an executing or to-be-executed process">
+      This resource represents a process executed by the compositor, or a
+      process which will be executed by the compositor shortly.
+
+      The client should transfer any appropriate state, then send an exec
+      request and await the exit event.
+
+      Once the compositor receives an exec request, it should fork, close all
+      file descriptors (except for the transferred ones), then dup2 them to the
+      appropriate file descriptor number. It should then exec (with PATH lookup,
+      e.g. via execlp or exevp) the requested command. Then it should send the
+      pid event, followed by the exit event when the child terminates.
+    </description>
+
+    <request name="environment">
+      <description summary="set an environment variable in the process">
+        Sets an environment variable for the new process. The compositor can
+        choose not to set any variables, including for example LD_PRELOAD.
+      </description>
+      <arg name="name" type="string" summary="environment variable name"/>
+      <arg name="value" type="string" summary="environment variable value"/>
+    </request>
+
+    <request name="fd">
+      <description summary="transfer a file descriptor to the process">
+        Transfers a file descriptor to the new process. The compositor will use
+        dup2 to assign the specified file descriptor number, then close the
+        danging fd.
+      </description>
+      <arg name="fd" type="fd" summary="file descriptor to transfer"/>
+      <arg name="num" type="uint" summary="first parameter to dup2"/>
+    </request>
+
+    <request name="exec">
+      <description summary="start the new process">
+        Asks the compositor to start the new process. Any request other than
+        destroy after this point is a protocol violation.
+      </description>
+    </request>
+
+    <event name="pid">
```

Comment:
As discussed in my comment about "exit", I don't really think this is necessary or that helpful, and there are other ways that this could be found out.